### ISSUE TITLE
[Strapi v5 RC] Fresh built Strapi cannot access content manager - Error: `useContentManagerInitData` must be used within `StrapiApp`

### ISSUE THREAD

**Issue by lung2008123** [strapi/strapi#21151]
### Required System information

- Node.js version: v18.20.4
- NPM version: 10.7.0
- Strapi version: Community 5.0.0-rc.0
- Database: postgres
- Operating system: Linux Ubuntu 24.04.1 LTS (Noble Numbat)
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Access the freshly built Strapi v5 content manager for the first time. The whole content manager page cannot load & display the error:
```
Error: `useContentManagerInitData` must be used within `StrapiApp`
```

### Steps to reproduce the behavior

1. Build a fresh Strapi v5 (5.0.0-rc.0) backend in AWS EC2  under production environment: `npm install && NODE_ENV=production npm run build`
2. Use pm2 to run `ecosystem.config.js`
3. Register first admin user
4. Access content manager for the first time & trigger the error below

### Expected behavior

Should be able to access content manager.

### Screenshots

![Screenshot 2024-09-05 at 10 05 49](https://github.com/user-attachments/assets/5041c31a-9495-45dc-a201-f54bd59d6fd6)

### Code snippets
package.json
```
{
  "name": "MY_PROJECT_NAME",
  "private": true,
  "version": "0.1.0",
  "description": "A Strapi application",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi",
    "copy-types": "node copyTypes.ts"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "typescript": "^5"
  },
  "dependencies": {
    "@strapi/email": "5.0.0-rc.0",
    "@strapi/plugin-cloud": "5.0.0-rc.0",
    "@strapi/plugin-users-permissions": "5.0.0-rc.0",
    "@strapi/provider-upload-aws-s3": "^4.25.0",
    "@strapi/strapi": "5.0.0-rc.0",
    "pg": "8.8.0",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-router-dom": "^6.0.0",
    "styled-components": "^6.0.0"
  },
  "author": {
    "name": "A Strapi developer"
  },
  "strapi": {
    "uuid": "MY_STRAPI_V5_UUID"
  },
  "engines": {
    "node": ">=18.0.0 <=20.x.x",
    "npm": ">=6.0.0"
  },
  "license": "MIT"
}
```

Error log
```
# used in production site
# built Strapi with NODE_ENV=production STRAPI_ENFORCE_SOURCEMAPS=true npm run build
# no source map is provided under production environment

Error: `useContentManagerInitData` must be used within `StrapiApp`
    at https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:4500:38188
    at GSe (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:2301:5033)
    at r (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:4500:38155)
    at ve (https://MY_ADMIN_DOMAIN/admin/layout-DHe2GdT4-BTG9PJRg.js:18:2286)
    at Ge (https://MY_ADMIN_DOMAIN/admin/layout-DHe2GdT4-BTG9PJRg.js:18:4319)
    at MP (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:45:17071)
    at J6e (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:47:44240)
    at Q6e (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:47:39940)
    at JZe (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:47:39863)
    at CL (https://MY_ADMIN_DOMAIN/admin/strapi-BC-9SC8Y.js:47:39712)
```
      
### Solution

Upgrade to Strapi v5 `v5.0.0-rc.13` can fix this access bug.

Replace `5.0.0-rc.0` by `v5.0.0-rc.13` & re-build the production Strapi by `npm install && NODE_ENV=production npm run build` command.

However, there might be a regression bug persist in `5.0.0-rc.0` that is worth-fixing.

**Comment by michsko544** [comment]
I have same problem every time I update version of RC.

I am using pnpm and usually when I delete pnpm-lock.yaml and reinstall packages `pnpm i` it fixes the problem.

**Comment by insivika** [comment]
I'm having the same issue. I'm on version `5.0.1`.

**Comment by minemos** [comment]
same here

**Comment by genesis-1** [comment]
```
Error: `useContentManagerInitData` must be used within `StrapiApp`
    at com/admin/strapi-sVcXgX2a.js:2438:6808
    at Lle (com/admin/strapi-sVcXgX2a.js:2283:4926)
    at i (hcom/admin/strapi-sVcXgX2a.js:2438:6775)
    at Fe (com/admin/layout-2CfjL0T9-D5gY8E9S.js:28:1605)
    at Qe (com/admin/layout-2CfjL0T9-D5gY8E9S.js:28:3638)
    at n$ (.com/admin/strapi-sVcXgX2a.js:40:17031)
    at vX (.com/admin/strapi-sVcXgX2a.js:42:44091)
    at pX (com/admin/strapi-sVcXgX2a.js:42:39809)
    at $Re (.com/admin/strapi-sVcXgX2a.js:42:39735)
    at _S (.com/admin/strapi-sVcXgX2a.js:42:39585)
```

**Comment by genesis-1** [comment]
> Same Issue as well.
> 
> ```
> Error: `useContentManagerInitData` must be used within `StrapiApp`
>     at com/admin/strapi-sVcXgX2a.js:2438:6808
>     at Lle (com/admin/strapi-sVcXgX2a.js:2283:4926)
>     at i (hcom/admin/strapi-sVcXgX2a.js:2438:6775)
>     at Fe (com/admin/layout-2CfjL0T9-D5gY8E9S.js:28:1605)
>     at Qe (com/admin/layout-2CfjL0T9-D5gY8E9S.js:28:3638)
>     at n$ (.com/admin/strapi-sVcXgX2a.js:40:17031)
>     at vX (.com/admin/strapi-sVcXgX2a.js:42:44091)
>     at pX (com/admin/strapi-sVcXgX2a.js:42:39809)
>     at $Re (.com/admin/strapi-sVcXgX2a.js:42:39735)
>     at _S (.com/admin/strapi-sVcXgX2a.js:42:39585)
> ```

i upgraded to v5.0.3 from v5.0.2 and the issue was fixed.

**Comment by michsko544** [comment]
I found a temporary solution. For example, if you are using `Strapi 5.0.5` like me you need to add the following `overrides` to `package.json`.

```json
"overrides": {
    "@strapi/admin": "5.0.5",
    "@strapi/database": "5.0.5",
    "@strapi/generators": "5.0.5",
    "@strapi/logger": "5.0.5",
    "@strapi/permissions": "5.0.5",
    "@strapi/types": "5.0.5",
    "@strapi/typescript-utils": "5.0.5",
    "@strapi/utils": "5.0.5"
  },
```

I think the problem is caused by installing different versions of Strapi packages in `node_modules` since Strapi uses `workspace:*` as a version tag (#20929). For example, the problem always occurred for me after a new version of Strapi was released. When my pipeline built an application in which I did not update Strapi to the latest one, the problem occurred.

I took a look at the `pnpm-lock.yaml` file and noticed that the `@strapi/admin@5.0.6` package was installed when I was using Strapi 5.0.5. After adding `overrides` the problem disappeared. Probably setting `@strapi/strapi@latest` will also solve the problem, but I prefer to have a fixed Strapi version, due to more control.

**Comment by dzakki** [comment]
I'm having the same issue. I'm on version on 5.0.2

**Comment by frontBOI** [comment]
@michsko544 's solution did work for me.
Created [an issue](https://github.com/strapi/strapi/issues/21982) related to the same problem.

**Comment by derrickmehaffy** [comment]
I'm bumping this up to high, we can't have people's app break every time we release an update.

**Comment by innerdvations** [comment]
The only report I'm seeing without mentioning an RC is https://github.com/strapi/strapi/issues/21982 and I'm trying to determine what could be the cause since I can't reproduce. If there is anyone who did not ever use an RC version of Strapi, can you provide more information?

- package manager being used
- initial strapi version
- target strapi version
- if you've tried any of the fixes and if they've worked or not
- if you have any plugins installed (including @strapi plugins)
- any other info such as using pm2 like the original issue

**Comment by killthekitten** [comment]
@innerdvations There's this PR in the preview plugin that perfectly reproduces the issue in our codebase when I add the package as a local plugin:

https://github.com/mattmilburn/strapi-plugin-preview-button/pull/140

My answers:
- package manager being used: yarn v1.22.19, node v18.20.0
- initial strapi version: latest v4 (not sure which that one was)
- target strapi version: 5.3.0
- if you've tried any of the fixes and if they've worked or not: haven't tried much

We are running on yarn v1.22.19 and node v18.20.0 with docker and without it. To connect the plugin I did this:

- Copied the plugin from the PR into `src/plugins/strapi-plugin-preview-button`
- Added it to plugins.js as `{ enabled: true, resolve: './src/plugins/strapi-plugin-preview-button' }`

If that's not enough for reproduction, I can try to create a separate repo.

Main package.json:
```json
{
  "name": "strapi",
  "version": "0.1.0",
  "private": true,
  "description": "A Strapi application",
  "license": "MIT",
  "scripts": {
    "build": "strapi build",
    "copy-types": "node src/scripts/copyTypes.js",
    "deploy": "strapi deploy",
    "develop": "strapi develop",
    "export-types": "yarn generate-types && yarn copy-types",
    "generate-types": "strapi ts:generate-types --debug",
    "postinstall": "patch-package",
    "patch-locales": "patch-package @strapi/plugin-i18n",
    "start": "strapi start",
    "strapi": "strapi"
  },
  "dependencies": {
    "@getpopsure/public-models": "^1.4.4",
    "@strapi/design-system": "^2.0.0-rc.12",
    "@strapi/plugin-cloud": "5.3.0",
    "@strapi/plugin-seo": "2.0.0",
    "@strapi/provider-email-sendgrid": "^5.3.0",
    "@strapi/strapi": "5.3.0",
    "@udx/strapi-provider-upload-google-cloud-storage": "^5.0.0",
    "patch-package": "^8.0.0",
    "pg": "^8.12.0",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-router-dom": "^6.0.0",
    "strapi-plugin-multi-select": "^2.1.1",
    "strapi-v5-plugin-populate-deep": "^4.0.4",
    "styled-components": "^6.0.0"
  },
  ...
  "engines": {
    "node": ">=18.0.0 <=20.x.x",
    "npm": ">=6.0.0"
  },
  ...
 
}
```

We also have another local plugin that hasn't been working correctly. I'm pretty sure that when I tested the preview-button plugin we didn't have this one around, but I'll share the deps of this one anyway:
```
{
  "version": "0.0.0",
  "keywords": [],
  "type": "commonjs",
  "exports": {
    "./package.json": "./package.json",
    "./strapi-admin": {
      "source": "./admin/src/index.js",
      "import": "./dist/admin/index.mjs",
      "require": "./dist/admin/index.js",
      "default": "./dist/admin/index.js"
    },
    "./strapi-server": {
      "source": "./server/src/index.js",
      "import": "./dist/server/index.mjs",
      "require": "./dist/server/index.js",
      "default": "./dist/server/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "strapi-plugin build",
    "watch": "strapi-plugin watch",
    "watch:link": "strapi-plugin watch:link",
    "verify": "strapi-plugin verify"
  },
  "dependencies": {
    "@strapi/design-system": "^2.0.0-rc.12",
    "@strapi/icons": "^2.0.0-rc.12",
    "react-intl": "^6.8.7"
  },
  "devDependencies": {
    "@strapi/strapi": "^5.3.0",
    "@strapi/sdk-plugin": "^5.2.7",
    "prettier": "^3.3.3",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-router-dom": "^6.27.0",
    "styled-components": "^6.1.13"
  },
  "peerDependencies": {
    "@strapi/strapi": "^5.3.0",
    "@strapi/sdk-plugin": "^5.2.7",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-router-dom": "^6.27.0",
    "styled-components": "^6.1.13"
  },
  "strapi": {
    "kind": "plugin",
    "name": "questionid-selector",
    "displayName": "questionid-selector",
    "description": "Internal custom field to select a question id"
  },
  "name": "questionid-selector",
  "description": "Internal custom field to select a question id",
  "license": "MIT",
  "author": "Feather team"
}
```

**Comment by innerdvations** [comment]
That plugin appears to be using an RC version of Strapi too. There seems to be a problem upgrading from any rc version to any release version, or when using a plugin that includes an rc version of Strapi.

**Comment by skywalker282** [comment]
> I found a temporary solution. For example, if you are using `Strapi 5.0.5` like me you need to add the following `overrides` to `package.json` and reinstall packages `pnpm install`.
> 
> ```json
> "overrides": {
>     "@strapi/admin": "5.0.5",
>     "@strapi/database": "5.0.5",
>     "@strapi/generators": "5.0.5",
>     "@strapi/logger": "5.0.5",
>     "@strapi/permissions": "5.0.5",
>     "@strapi/types": "5.0.5",
>     "@strapi/typescript-utils": "5.0.5",
>     "@strapi/utils": "5.0.5"
>   },
> ```
> 
> I think the problem is caused by installing different versions of Strapi packages in `node_modules` since Strapi uses `workspace:*` as a version tag (#20929). For example, the problem always occurred for me after a new version of Strapi was released. When my pipeline built an application in which I did not update Strapi to the latest one, the problem occurred.
> 
> I took a look at the `pnpm-lock.yaml` file and noticed that the `@strapi/admin@5.0.6` package was installed when I was using Strapi 5.0.5. After adding `overrides` the problem disappeared. Probably setting `@strapi/strapi@latest` will also solve the problem, but I prefer to have a fixed Strapi version, due to more control.

This worked for me on v5.0.5

**Comment by hanpaine** [comment]
Hi folks 👋 we are still unable to reproduce this issue after extensive investigations. It appears to consistently appear with references between RCs and Stable releases, but not on clean builds. If you do encounter it, please use @michsko544 's [workaround](https://github.com/strapi/strapi/issues/21151#issuecomment-2411667883) (thank you 🙏 ) or open a new Issue with as many reproduction details as possible.

As an aside, we are currently fixing a problem on #22162 that might be relevant for some folks here. Thanks everyone for their contributions and ideas, we really appreciate the engagement 🚀

**Comment by jigyarth-searce** [comment]
> ```json
> "overrides": {
>     "@strapi/admin": "5.0.5",
>     "@strapi/database": "5.0.5",
>     "@strapi/generators": "5.0.5",
>     "@strapi/logger": "5.0.5",
>     "@strapi/permissions": "5.0.5",
>     "@strapi/types": "5.0.5",
>     "@strapi/typescript-utils": "5.0.5",
>     "@strapi/utils": "5.0.5"
>   },
> ```

Works for me thanks

**Comment by michsko544** [comment]
@hanpaine In Strapi 5.6.0, the problem still persists

**Comment by michsko544** [comment]
### Steps to Reproduce
Add the following dependencies to your package.json:
```json
{
  "@strapi/email": "5.6.0",
  "@strapi/plugin-graphql": "5.6.0",
  "@strapi/plugin-users-permissions": "5.6.0",
  "@strapi/provider-upload-local": "5.6.0",
  "@strapi/strapi": "5.6.0"
}
```
Run npm install.
Generate a complete dependency tree to confirm that some packages are resolved to version 5.7.0.
`npm ls --all | grep "@strapi/*"`

### Findings
The following packages in version 5.6.0 pull in dependencies from 5.7.0:

- `@strapi/email@5.6.0`
Installs: @strapi/admin@5.7.0
Dependencies: @strapi/data-transfer@5.7.0, @strapi/permissions@5.7.0, @strapi/utils@5.7.0, etc.
- `@strapi/content-manager@5.6.0`
Installs: @strapi/admin@5.7.0
- `@strapi/content-releases@5.6.0`
Installs: @strapi/admin@5.7.0
- `@strapi/content-type-builder@5.6.0`
Installs: @strapi/admin@5.7.0
- `@strapi/i18n@5.6.0`
Installs: @strapi/admin@5.7.0
- `@strapi/review-workflows@5.6.0`
Installs: @strapi/admin@5.7.0
- `@strapi/upload@5.6.0`
Installs: @strapi/admin@5.7.0

### Recommendations for maintainers
1. Audit Dependencies
Review the package.json files of all @strapi packages in version 5.6.0 to identify where dependencies are pinned to ^5.x.x or ~5.x.x, which could allow unintended upgrades to 5.7.0.

2. Test Dependency Resolution
Use tools like [npm dedupe](https://docs.npmjs.com/cli/v9/commands/npm-dedupe) or manually simulate an install with strict version resolutions to ensure no mismatches occur during the installation process.

3. Enhance Documentation
Provide detailed guidance on how developers can avoid version mismatches when installing Strapi packages, including recommendations for lock files, resolution strategies, and tools like npm-force-resolutions.

4. Consider Publishing Meta Packages
Create meta packages for major versions (e.g., @strapi/core@5.6) that pre-bundle compatible versions of all sub-packages, ensuring consistency out of the box.

5. Automated Testing for Version Compatibility
Introduce automated testing in the CI/CD pipeline to validate that all dependencies for a given version (e.g., 5.6.0) resolve correctly without pulling in newer versions.

This issue affects developers attempting to use Strapi older versions, leading to broken builds due to mismatched versions.

Please reopen this issue to avoid creating another duplication.

*Dependency analysis by ChatGPT

**Comment by nextlevelshit** [comment]
Same happend with my project, when I've first added `@strapi/mail` or `@strapi/provider-email-nodemailer`. Looks like it is important to have all strapi packages in sync:

```ts
  "dependencies": {
    "@strapi/email": "^5.11.0",
    "@strapi/plugin-cloud": "^5.11.0",
    "@strapi/plugin-documentation": "^5.11.0",
    "@strapi/plugin-users-permissions": "^5.11.0",
    "@strapi/provider-email-nodemailer": "^5.11.0",
    "@strapi/strapi": "^5.11.0",
    "nodemailer": "^6.10.0",
    ...
  },
```
